src/Microsoft.Identity.Web/DownstreamApiSupport/DownstreamWebApi.cs

```diff
@@ -0,0 +1,172 @@
+        {
+            CalledApiOptions effectiveOptions = MergeOptions(optionsInstanceName, calledApiOptionsOverride);
+
+            // verify scopes is not null
+            if (string.IsNullOrEmpty(effectiveOptions.Scopes))
+            {
+                throw new ArgumentException("Scopes need to be passed-in either by configuration or by the delegate overring it.");
+            }
+
+            string accessToken = await _tokenAcquisition.GetAccessTokenForUserAsync(effectiveOptions.GetScopes(), effectiveOptions.Tenant)
+                .ConfigureAwait(false);
+
+            HttpResponseMessage response;
+            using (HttpRequestMessage httpRequestMessage = new HttpRequestMessage(effectiveOptions.HttpMethod, effectiveOptions.GetApiUrl()))
+            {
+                if (content != null)
+                {
+                    httpRequestMessage.Content = content;
+                }
+
+                httpRequestMessage.Headers.Add("Authorization", $"bearer {accessToken}");
+                response = await _httpClient.SendAsync(httpRequestMessage).ConfigureAwait(false);
+            }
+
+            return response;
+        }
+
+        /// <summary>
+        /// Merge the options from configuration and override from caller.
+        /// </summary>
+        /// <param name="optionsInstanceName">Named configuration.</param>
+        /// <param name="calledApiOptionsOverride">Delegate to override the configuration.</param>
+        internal /* for tests */ CalledApiOptions MergeOptions(
+            string optionsInstanceName,
+            Action<CalledApiOptions>? calledApiOptionsOverride)
+        {
+            // Gets the options from configuration (or default value)
+            CalledApiOptions options;
+            if (optionsInstanceName != null)
+            {
+                options = _namedOptions.Get(optionsInstanceName);
+            }
+            else
+            {
+                options = _namedOptions.CurrentValue;
+            }
+
+            // Give a chance to the called to override defaults for this call
+            CalledApiOptions clonedOptions = options.Clone();
+            calledApiOptionsOverride?.Invoke(clonedOptions);
+            return clonedOptions;
+        }
+
+        /// <inheritdoc/>
+        public async Task<TOutput> CallWebApiForUserAsync<TInput, TOutput>(
+            string optionsInstanceName,
+            TInput input,
+            Action<CalledApiOptions>? calledApiOptionsOverride,
+            ClaimsPrincipal? user)
+            where TOutput : class
+        {
+            StringContent? jsoncontent;
+            if (input != null)
+            {
+                var jsonRequest = JsonSerializer.Serialize(input);
+                jsoncontent = new StringContent(jsonRequest, Encoding.UTF8, "application/json");
+// case of patch?               jsoncontent = new StringContent(jsonRequest, Encoding.UTF8, "application/json-patch+json");
+            }
+            else
```

Comment:
I think this `else` is unnecessary, `jsoncontent` would still be null here.

Reply:
I tried but the compiler was not happy line 131.